src/app/design-model-module/service/design-model.service.ts

```diff
@@ -0,0 +1,162 @@
+        patternContainer.patterns = designModel._embedded.patterns;
+        patternContainer._links = designModel._links;
+        return patternContainer;
+      })
+    );
+  }
+
+  getPatternContainer(url: string): Observable<PatternContainer> {
+    return this.getDesignModelByUrl(url).pipe(
+      map(designModel => {
+        const patternContainer = new PatternContainer();
+        patternContainer.patterns = designModel._embedded.patterns;
+        patternContainer._links = designModel._links;
+        return patternContainer;
+      })
+    );
+  }
+
+  getPatternContainerByUuid(uuid: string): Observable<PatternContainer> {
+    return this.getDesignModelByUuid(uuid).pipe(
+      map(designModel => {
+        const patternContainer = new PatternContainer();
+        patternContainer.patterns = designModel._embedded ? designModel._embedded.patterns : [];
+        patternContainer._links = designModel._links;
+        return patternContainer;
+      })
+    );
+  }
+
+  getDesignModelByUrl(url: string): Observable<DesignModel> {
+    return this.http.get<DesignModel>(url);
+  }
+
+  getDesignModelByUuid(uuid: string): Observable<DesignModel> {
+    return this.http.get<DesignModel>(this.repoEndpoint + '/designModels/' + uuid + '/patterns');
+  }
+
+  createLink(
+    url,
+    edge: DirectedEdgeModel | UndirectedEdgeModel | AddDirectedEdgeToViewRequest | AddUndirectedEdgeToViewRequest
+  ): Observable<HttpResponse<Object>> {
+    return this.http.post(url, edge, { observe: 'response' });
+  }
+
+  addLinks(patternContainer: PatternContainer, items: LinksToOtherPattern[]): Observable<any> {
+
+    const observables = items
+      .map(item => item.type === 'directed' ?
+        this.http.post(patternContainer._links.directedEdges.href, new AddDirectedEdgeToViewRequest(<DirectedEdgeModel>item.edge), { observe: 'response' }) :
+        this.http.post(
+          patternContainer._links.undirectedEdges.href,
+          new AddUndirectedEdgeToViewRequest(<UndirectedEdgeModel>item.edge),
+          { observe: 'response' }
+        )
+      );
+    return observables.length > 0 ? forkJoin(observables) : EMPTY;
+  }
+
+  getDirectedEdges(patternContainer: PatternContainer): Observable<Embedded<DirectedEdesResponse>> {
+    return this.http.get<Embedded<DirectedEdesResponse>>(patternContainer._links.directedEdges.href);
+  }
+
+  getUndirectedEdges(patternContainer: PatternContainer): Observable<Embedded<UndirectedEdesResponse>> {
+    return this.http.get<Embedded<UndirectedEdesResponse>>(patternContainer._links.undirectedEdges.href);
+  }
+
+  deleteLink(patternLink: any): Observable<any> {
+    return this.http.delete(patternLink);
+  }
+
+  saveGraph(patternContainer: PatternContainer, nodeList: any[]) {
+    return this.http.post<any>(patternContainer._links.patterns.href, nodeList, { observe: 'response' });
+  }
+
+  getGraph(patternContainer: PatternContainer): Observable<{ graph: Array<GraphNode> }> {
```

Comment:
Was the functionality to retrieve a previously saved graph (including positions of nodes) moved or deleted?